File: 2-ad-hoc/1147.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
//
int main(){
    char*inp;
    int**tabuleiro,*cavalo,*movis;
    for(int cont=1;;++cont){
        for(int i=0;i<9;++i){
            inp=(char*)malloc(3*sizeof(char));
            scanf("%s",inp);
            if(inp[0]=='0')
                break;
            if(i==0){
                cavalo=(int*)malloc(2*sizeof(int));
                cavalo[0]=inp[0]-'0'-1;
                cavalo[1]=(int)inp[1]-97;
                tabuleiro=(int**)malloc(8*sizeof(*tabuleiro));
                for(int j=0;j<8;++j){
                    tabuleiro[j]=(int*)calloc(8,sizeof(**tabuleiro));
                }
            } else if(inp[0]-'0'>1){
                (int)inp[1]-96<=7?tabuleiro[inp[0]-'0'-2][(int)inp[1]-96]=1:tabuleiro[0][0];
                (int)inp[1]-98>=0?tabuleiro[inp[0]-'0'-2][(int)inp[1]-98]=1:tabuleiro[0][0];
            }
            free(inp);
        }
        if(inp[0]=='0')
            break;
        movis=(int*)calloc(1,sizeof(*movis));
        for(int i=0;i<2;++i){
            if(cavalo[0]+2-i<=7){
                (tabuleiro[cavalo[0]+2-i][cavalo[1]+1+i]==0&&cavalo[1]+1+i<=7)?++*movis:*movis;
                (tabuleiro[cavalo[0]+2-i][cavalo[1]-1-i]==0&&cavalo[1]-1-i>=0)?++*movis:*movis;
            }
            if(cavalo[0]-2+i>=0){
                (tabuleiro[cavalo[0]-2+i][cavalo[1]+1+i]==0&&cavalo[1]+1+i<=7)?++*movis:*movis;
                (tabuleiro[cavalo[0]-2+i][cavalo[1]-1-i]==0&&cavalo[1]-1-i>=0)?++*movis:*movis;
            }
        }
        printf("Caso de Teste #%i: %i movimento(s).\n",cont,*movis);
        free(movis);
        free(tabuleiro);
    }
    return 0;
}

File: 2-ad-hoc/1147.py
cont=0
matrix=[]
cavalo=None
epa=1
while True:
    x=input()
    if x=='0':
        break
    if cont==0:
        cavalo=[int(x[0])-1,ord(x[1])-97]
        for i in range(8):
            matrix.append([0,0,0,0,0,0,0,0])
    elif(x[0]!="0"):
        if ord(x[1])-96<=7:matrix[int(x[0])-2][ord(x[1])-96]=1
        if ord(x[1])-98>=0:matrix[int(x[0])-2][ord(x[1])-98]=1
    if cont==8:
        movi=0
        for i in range(2):
            if cavalo[0]+2-i<=7:
                if cavalo[1]+1+i<=7: movi+=1 if matrix[cavalo[0]+2-i][cavalo[1]+1+i]==0 else 0
                if cavalo[1]-1-i>=0: movi+=1 if matrix[cavalo[0]+2-i][cavalo[1]-1-i]==0 else 0
            if cavalo[0]-2+i>=0:
                if cavalo[1]+1+i<=7: movi+=1 if matrix[cavalo[0]-2+i][cavalo[1]+1+i]==0 else 0
                if cavalo[1]-1-i>=0: movi+=1 if matrix[cavalo[0]-2+i][cavalo[1]-1-i]==0 else 0
        print('Caso de Teste #{}: {} movimento(s).'.format(epa,movi))
        cont=-1
        epa+=1
        matrix.clear()
        cavalo=None
    cont+=1